app/services/process_payment_intent.rb

```diff
@@ -36,10 +36,12 @@ def call!
     process_payment
 
     if payment.reload.completed?
-      payment.mark_as_processed
+      payment.completed_authorization
+      payment.clear_authorization_url
 
       Result.new(ok: true)
     else
+      payment.failed_authorization
```

Comment:
This will transition it to `failed` state? The payment object will behave differently after we do this, right? Is this what we want?

Reply:
I think it is - if the customer denies authorization on the extra step, we'd treat the payment the same as if it failed for any other reason. They'd have to create a new payment in order to pay the balance on the order.